### PR TITLE
tbb: Fix building on clangarm64

### PR DESCRIPTION
**Pull request by MehdiChinoune**
@hmartinez82

### CODE REVIEW

File: mingw-w64-tbb/004-support-mingw-aarch64.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/src/tbbmalloc/TypeDefinitions.h
++++ b/src/tbbmalloc/TypeDefinitions.h
+@@ -25,7 +25,7 @@
+ #       define __ARCH_ipf 1
+ #   elif defined(_M_IX86)||defined(__i386__) // the latter for MinGW support
+ #       define __ARCH_x86_32 1
+-#   elif defined(_M_ARM) || defined(_M_ARM64)
++#   elif defined(_M_ARM) || defined(_M_ARM64) || defined(__aarch64__) // the latter for MinGW support
```
Comment (Biswa96) [inline mingw-w64-tbb/004-support-mingw-aarch64.patch:8]: Theoretically, this patch should not be required. mingw-w64 defines `_M_ARM64` if `__aarch64__` is defined. See https://github.com/mingw-w64/mingw-w64/blob/master/mingw-w64-headers/crt/_mingw_mac.h

Comment (jeremyd2019) [reply mingw-w64-tbb/004-support-mingw-aarch64.patch:8]: Probably hasn't included anything that ultimately includes _mingw_mac.h at that point.  I've run into that once or twice when checking for `_UCRT`

Comment (MehdiChinoune) [reply mingw-w64-tbb/004-support-mingw-aarch64.patch:8]: Then, why does it show this https://github.com/oneapi-src/oneTBB/blob/v2021.5.0/src/tbbmalloc/TypeDefinitions.h#L31

Comment (Biswa96) [reply mingw-w64-tbb/004-support-mingw-aarch64.patch:8]: OK, Jeremy is right. For example, the following code fails in x86_64

```c
#ifdef _M_X64
#warning "right"
#else
#error "wrong"
#endif

#include <stdio.h>

int main(){}
```

But if `stdio.h` is included at first things are fine.

**Comment by lazka** [comment]
should we just try in autobuild?

**Comment by MehdiChinoune** [comment]
> should we just try in autobuild?

Maybe1

**Pull request merged**